src/xma/src/xmaapi/xmafilter.cpp

```diff
@@ -203,7 +204,7 @@ xma_filter_session_create(XmaFilterProperties *filter_props)
     }
 
     //Obtain lock only for a) singleton changes & b) kernel_info changes
-    std::unique_lock<std::mutex> guard1(g_xma_singleton->m_mutex);
+    std::unique_lock<std::shared_timed_mutex> guard1(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
    std::unique_lock guard1(g_xma_singleton->m_mutex);
```

Reply:
template type deduction not available in c++14